OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -892,6 +892,14 @@ internal static void UpgradeActorRules(int engineVersion, ref List<MiniYamlNode>
 					}
 				}
 
+				if (engineVersion < 20150501)
+				{
+					// Made buildings grant prerequisites explicitly
+					if (depth == 0 && node.Value.Nodes.Exists(n => n.Key == "Inherits" &&
+						(n.Value.Value == "^Building" || n.Value.Value == "^BaseBuilding")))
+						node.Value.Nodes.Add(new MiniYamlNode("ProvidesCustomPrerequisite@buildingName", ""));
```

Comment:
I am and see no reason not to, but if it goes against convention then it shouldn't be used.

Either way `buildingName` is bogus.